service/controller/app/v1/resource/chartstatus/create_test.go

```diff
@@ -0,0 +1,243 @@
+package chartstatus
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned/fake"
+	"github.com/giantswarm/micrologger/microloggertest"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/kubeconfig"
+)
+
+func TestResource_EnsureCreated(t *testing.T) {
+	tests := []struct {
+		name         string
+		obj          *v1alpha1.App
+		chart        *v1alpha1.Chart
```

Comment:
It would be good to add the App status to the tests struct. So you can check that the app status is set correctly.